rust_dev_preview/cross_service/rest_ses/src/work_item/repository.rs

```diff
@@ -0,0 +1,206 @@
+        Some(item) => Ok(item),
+        None => Err(WorkItemError::Other(
+            "Somehow len() == 1 but get(0) is None".to_string(),
+        )),
+    }?;
+
+    Ok(item.to_owned())
+}
+
+/// Retrieve a list of all records with a given WorkItemArchived state.
+pub async fn list(
+    archive: WorkItemArchived,
+    client: &RdsClient,
+) -> Result<Vec<WorkItem>, WorkItemError> {
+    let statement = client.execute_statement();
+    let statement = match archive {
+        WorkItemArchived::All => statement.sql(format!(r#"SELECT {FIELDS} FROM Work;"#)),
+        _ => statement
+            .sql(format!(
+                r#"SELECT {FIELDS} FROM Work WHERE archive = :archive ;"#
+            ))
+            .set_parameters(params![("archive", format!("{}", u8::from(&archive)))]),
+    };
+
+    let statement = statement
+        .format_records_as(RecordsFormatType::Json)
+        .send()
+        .await;
+
+    parse_rds_output(statement)
+}
+
+/// Update a single item in the database, by ID.
+/// Retrieves the value after update for its result.
+#[tracing::instrument(name = "Repository Update WorkItem", skip(client))]
+pub async fn update(item: &WorkItem, client: &RdsClient) -> Result<WorkItem, WorkItemError> {
+    client
+        .execute_statement()
+        .sql(
+            r#"
+            UPDATE Work
+            SET 
+                username = :username,
+                date = :date,
+                description = :description,
+                guide = :guide,
+                status = :status,
+                archive = :archive
+            WEHRE idwork = :idwork;
+        "#,
+        )
+        .set_parameters(params![
+            ("idwork", item.idwork),
+            ("username", item.name),
+            ("date", format!("{}", item.date.format(RDS_DATE_FORMAT))),
+            ("description", item.description),
+            ("guide", item.guide),
+            ("status", item.status),
+            ("archive", format!("{}", u8::from(&item.archive)))
+        ])
+        .send()
+        .await
+        .map_err(|err| {
+            tracing::error!("Failed to update user: {id} {err:?}", id = item.idwork);
+            WorkItemError::RDSError(err.into())
+        })?;
+    retrieve(item.idwork().to_string(), client).await
+}
+
+/// Delete an item from the database, returning () on success.
+#[tracing::instrument(name = "Repository Delete WorkItem", skip(client))]
+pub async fn delete(id: String, client: &RdsClient) -> Result<(), WorkItemError> {
+    client
+        .execute_statement()
+        .sql(r#"DELETE FROM Work WHERE idwork = :idwork"#)
+        .set_parameters(params![("idwork", id)])
+        .send()
+        .await
+        .map(|_| Ok(()))
+        .map_err(|err| {
+            tracing::error!("Failed to delete user: {id} {err:?}");
+            WorkItemError::RDSError(err.into())
+        })?
+}
+
+/// Attempt to parse an Amazon Relational Database Service (Amazon RDS) Data SQL statement to a WorkItem.
```

Comment:
I think the full name has to be: Amazon Relational Database Service (Amazon RDS) Data Service

Reply:
Liz corrected it to this?